assets/js/util.ts

```diff
@@ -196,3 +197,23 @@ export const sortScreens = (screenList: Screen[]) => {
     screenTypeOrder.indexOf(a.type) >= screenTypeOrder.indexOf(b.type) ? 1 : -1
   );
 };
+
+export const sortByStationOrder = (
+  places: Place[],
+  filteredLine: string,
+  reverse?: boolean
+) => {
+  const stationOrder = STATION_ORDER_BY_LINE[filteredLine.toLowerCase()];
+
+  places.sort((placeA, placeB) => {
+    const indexA = stationOrder.findIndex((station) => {
+      return station.name.toLowerCase() === placeA.name.toLowerCase();
+    });
+    const indexB = stationOrder.findIndex((station) => {
+      return station.name.toLowerCase() === placeB.name.toLowerCase();
+    });
+    return indexA - indexB;
+  });
```

Comment:
This is pretty inefficient since every time we need to compare two elements of `places` (potentially once for every unique pair of elements—`O(n^2)`), we need to iterate through `stationOrder` twice. That's something like cubic time—`O(m*n^2)` where m is `stationOrder.length` and n is `places.length`—in the worst case.

It should be faster to compute the indices once for each place, up front. You can also create a map from `stationOrder` to make lookups faster. What we really want is something like `Enum.sort_by/2`, but that method isn't available in standard JS so we can do it ourselves. _Or_, add lodash to the project and use [its `sortBy` function](https://lodash.com/docs/4.17.15#sortBy).

```ts
const stationOrderToIndex = Object.from_entries(
  stationOrder.map((station, i) => [station.name.toLowerCase(), i])
);

const placesByStationOrder = places.map((place) => ({
  place,
  index: stationOrderToIndex[place.name.toLowerCase()]
}));

const sortedPlaces = placesByStationOrder
                       .sort(({index: i1}, {index: i2}) => i1 - i2) 
                       .map(({place}) => place);

return reverse? sortedPlaces.reverse() : sortedPlaces;
```
(I didn't write this in vscode, so please double check that it works if you decide to use it!)

Reply:
Wow, great catch. This is working great as is so I'll stick with this instead of adding `lodash` at the moment.